### DISCUSSION TITLE
How to setup an automation to trigger a card, track from your library

### DISCUSSION THREAD

**Discussion by f1wade** [cdnninja/yoto_ha#119] [Q&A] [answered]
I have just found and installed this integration.
Looks promising, thanks for getting it out their.

I can see data from my yotos.

but I want to be able to see the library, and then pick a card and track and play it from HA.

anyone done this or able to help me figure it out please?

**Comment by cdnninja** [comment, marked as answer]
Click browse media within the media player entity. It should let you browse and play. You can also use the automation service call to pass the card id to an automation.

**Reply by f1wade** [reply]
I can browse media and see all in the library.
But can't see the ids for any of them

And when I play them the yoto is still saying idle in ha, even after a few mins.

**Reply by cdnninja** [reply]
Odd!  To get the id view the media player in the developer menu.   

Not sure why it doesn't show playing though.

**Reply by f1wade** [reply]
Working now, just not waiting long enough.
Once I saw details in developer view, and template details I could figure the rest out for automations.

This allows alarms to turn on actual cards, which is brilliant.

Thanks so much for doing this.

**Comment by f1wade** [comment]
Answer is, when you add the media player entity, you can then see browse and pick what you want.
When playing it shows content I'd in developer state view, or media attributes on markdown.

Be nice to see the content ids when browsing, but not sure that possible?

**Reply by cdnninja** [reply]
I could but it means the titles would have that hanging off the end.  Does that make it too ugly for day to day use?

**Reply by f1wade** [reply]
For my system, its just me, so if seperated well and copy able, then would be fine.

But I don't know if I speak for all users...
Unless if could be a flag somewhere?

**Comment by mosschief** [comment]
One issue I'm running into is that when I trigger sleep sounds, it does not loop. It will just end after a couple minutes of playing. When triggered from the app it will play continuously. Any idea how to get around this?

**Reply by cdnninja** [reply]
I don't personally use sleep sounds. However, checked a few things.  When I enable debug logs and run it from within the app it sends this.  Key bit being repeatALL being off.  In the app it also shows playing without repeat enabled.   Does your show repeat? If so maybe that was turned on in the app and it remembers it? 

`2025-01-07 19:04:20.445 DEBUG (Thread-7 (_thread_main)) [yoto_api.YotoMQTTClient] yoto_api - MQTT Message: {"trackLength":1854,"position":0,"cardId":"gOBNG","repeatAll":false,"source":"card","cardUpdatedAt":"2022-04-14T16:59:22.445Z","chapterTitle":"Pink Noise","chapterKey":"Pink","trackTitle":"Pink Noise","trackKey":"Pink","streaming":true,"volume":5,"volumeMax":5,"playbackStatus":"playing","playbackWait":false,"sleepTimerActive":false,"eventUtc":1736301867}
`

If your app does show the repeat icon being highlighted I need to add the "repeat" feature to this integration.  You could then call play and enable repeat.

**Comment by mosschief** [comment]
My app does show the repeat icon highlighted
